src/azul/terra.py

```diff
@@ -296,15 +299,20 @@ def _request(self,
         timeout = config.terra_client_timeout
         log.debug('_request(%r, %r, fields=%r, headers=%r, timeout=%r, body=%r)',
                   method, url, fields, headers, timeout, body)
-        response = self._http_client.request(method,
-                                             url,
-                                             fields=fields,
-                                             headers=headers,
-                                             # FIXME: Service should return 503 response when Terra client times out
-                                             #        https://github.com/DataBiosphere/azul/issues/3968
-                                             timeout=timeout,
-                                             retries=False,
-                                             body=body)
+        try:
+            response = self._http_client.request(method,
+                                                 url,
+                                                 fields=fields,
+                                                 headers=headers,
+                                                 # FIXME: Service should return 503 response when Terra client times out
+                                                 #        https://github.com/DataBiosphere/azul/issues/3968
+                                                 timeout=timeout,
+                                                 retries=False,
+                                                 body=body)
+        except TimeoutError:
+            raise RequirementError(f'Did not recieve response from {url!r} '
```

Comment:
This is a stretch, a timeout is not a requirement error. I think we may want to add a dedicated `TerraClientException` hierarchy, with subclasses for the various cases.